rest/admin_api.go

```diff
@@ -1770,3 +1776,29 @@ func databaseLoadErrorAsHTTPError(err error) error {
 	}
 	return base.HTTPErrorf(http.StatusInternalServerError, "couldn't load database: %v", err)
 }
+
+// checkUserAPIReadOnlyFields will return true if read only user fields are unchanged, else false. If a read only field
+// is unchanged it will nil the field so the update to principle ignores it.
```

Comment:
> If a read only field is unchanged it will nil the field so the update to principle ignores it.

There is already logic inside `UpdatePrincipal` to skip if the non-nil values are the same, but there's no harm in being defensive here.

Reply:
My thinking here is due to this -> https://github.com/couchbase/sync_gateway/blob/9b782554ec47ef47bd2c19ed2f4965d662469a7e/db/users.go#L47 when I ran some test locally where a user is only granted channels, and thus only has issuer and channels defined on the PUT, it fails here. So thought it best to just nil them all in defense of that.